src/app/CommandResponseHelper.h

```diff
@@ -53,7 +53,7 @@ class CommandResponseHelper
 
     CHIP_ERROR Failure(Protocols::InteractionModel::Status aStatus)
     {
-        CHIP_ERROR err = mCommandHandler->AddStatus(mCommandPath, aStatus);
+        CHIP_ERROR err = mCommandHandler->FailableAddStatus(mCommandPath, aStatus);
```

Comment:
This is fake error handling. What the caller of this will do if mSentResponse is false is ... AddStatus.

So we should just AddStatus here and set mSentResponse to true unconditionally....

Reply:
I think I should only do so if we make Failure return void. since currently we claim this is a Fallible path since we return chip error, doing the falllible bit seems to apply.

I also think you are correct that these should just be void, however I would rather do that as a separate PR.

Reply:
Mostly if we don't use FallibleAddStatus here we can fix its signature to make sense for the CommandHandler-internal uses...  But yes, followup is fine for this.